components/search/src/filter.rs

```diff
@@ -217,4 +299,458 @@ mod tests {
+                order_hint: None,
+                telemetry_suffix: None,
+                urls: SearchEngineUrls {
+                    search: SearchEngineUrl {
+                        base: "https://example.com/3".to_string(),
+                        method: "GET".to_string(),
+                        params: Vec::new(),
+                        search_term_param_name: None,
+                    },
+                    suggestions: None,
+                    trending: None,
+                },
+            },
+            SearchEngineDefinition {
+                aliases: Vec::new(),
+                charset: "UTF-8".to_string(),
+                classification: SearchEngineClassification::General,
+                identifier: "engine4wildcardmatch".to_string(),
+                partner_code: String::new(),
+                name: "Test 4".to_string(),
+                order_hint: None,
+                telemetry_suffix: None,
+                urls: SearchEngineUrls {
+                    search: SearchEngineUrl {
+                        base: "https://example.com/4".to_string(),
+                        method: "GET".to_string(),
+                        params: Vec::new(),
+                        search_term_param_name: None,
+                    },
+                    suggestions: None,
+                    trending: None,
+                },
+            },
+        ]
+    });
+
+    #[test]
+    fn test_determine_default_engines_returns_global_default() {
+        let (default_engine_id, default_engine_private_id) = determine_default_engines(
+            &ENGINES_LIST,
+            Some(JSONDefaultEnginesRecord {
+                global_default: "engine2".to_string(),
+                global_default_private: "".to_string(),
+                specific_defaults: Vec::new(),
+            }),
+            &SearchUserEnvironment {
+                locale: "fi".into(),
+                region: "FR".into(),
+                update_channel: SearchUpdateChannel::Default,
+                distribution_id: String::new(),
+                experiment: String::new(),
+                app_name: SearchApplicationName::Firefox,
+                version: String::new(),
+            },
+        );
+
+        assert_eq!(
+            default_engine_id.unwrap(),
+            "engine2",
+            "Should have returned the global default engine"
+        );
+        assert!(
+            default_engine_private_id.is_none(),
+            "Should not have returned an id for the private engine"
+        );
+
+        let (default_engine_id, default_engine_private_id) = determine_default_engines(
+            &ENGINES_LIST,
+            Some(JSONDefaultEnginesRecord {
+                global_default: "engine2".to_string(),
+                global_default_private: String::new(),
+                specific_defaults: vec![JSONSpecificDefaultRecord {
+                    default: "engine1".to_string(),
+                    default_private: String::new(),
```

Comment:
Why this default_private is `String::new()` and the `global_default_private` above on line 395 is `"".to_string()`?

Could we keep these consistent and use one way to define the string, otherwise if there's a reason to use different methods in Rust -- I'm fine with that too. I wasn't sure why they're different. I'm wondering if it's because they are different types or something else?

Reply:
>  if there's a reason to use different methods in Rust -- I'm fine with that too

They're equivalent! 😊 And so are all these:

```
String::new()
"".to_string()
"".to_owned()
"".into() // When assigning to a `String`.
String::from("")
```

Reply:
Thank you both!